dipu/CMakeLists.txt

```
@@ -154,6 +154,11 @@ set_target_properties(
 target_include_directories(diopi_impl INTERFACE "${DIOPI_PROTO_PATH}/include")
 target_compile_definitions(diopi_impl INTERFACE DIOPI_ATTR_WEAK)
 
+include_directories(${PROJECT_SOURCE_DIR}/third_party/kineto/libkineto/third_party/fmt/include)
+include_directories(${PROJECT_SOURCE_DIR}/third_party/kineto/libkineto/include)
+include_directories(${PROJECT_SOURCE_DIR}/third_party/kineto/libkineto/src)
+add_definitions(-DUSE_KINETO)
```

然后参照 DIOPI，将 kineto import 进来，加强模块化
```suggestion
add_library(kineto_fmt STATIC IMPORTED)
add_dependencies(kineto_fmt kineto_build)
set_target_properties(kineto_fmt PROPERTIES
  IMPORTED_LOCATION "${KINETO_BUILD_DIR}/fmt/libfmt.a")
target_include_directories(kineto_fmt INTERFACE SYSTEM
  "${KINETO_SOURCE_DIR}/third_party/fmt/include")
add_library(kineto_lib STATIC IMPORTED)
add_dependencies(kineto_lib kineto_build)
set_target_properties(kineto_lib PROPERTIES
  IMPORTED_LOCATION "${KINETO_BUILD_DIR}/libkineto.a")
target_include_directories(kineto_lib INTERFACE SYSTEM
  ${KINETO_SOURCE_DIR}/include
  ${KINETO_SOURCE_DIR}/src
)
add_library(kineto INTERFACE)
target_link_linbraries(kineto INTERFACE kineto_fmt kineto_lib)
target_add_definitions(kineto INTERFACE USE_KINETO)
```

这两部分代码可以在 thirdparty 下面新建一个 CMake，以保持简洁

---

很好的建议，我下周五之前会再把这部分重构一下。

在thirdparty下面新建CMake，实现对整个thirdparty的编译，并能跟DIPU的编译选项设置保持一致，实现一体化编译。当前已经发现在某个厂商设备上由于_GLIBCXX_USE_CXX11_ABI设置不一样导致找不到fmt的符号。

---

方便的话可以在#346 里面帮我们维护一下这个计划的进展，因为我们最近一直在重构整个 build 系统

---

好的，我看你已经把这个事情记录进去了，有进展我会在 https://github.com/DeepLink-org/DIPU/issues/346 里面更新

---

手动点赞     这是review的典范